src/main/java/pro/sky/cookingappnew/controllers/FileController.java

```diff
@@ -0,0 +1,111 @@
+import pro.sky.cookingappnew.model.Ingridients;
+import pro.sky.cookingappnew.model.Recipe;
+import pro.sky.cookingappnew.services.ExportRecipeService;
+import pro.sky.cookingappnew.services.FileService;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.*;
+import java.util.HashMap;
+import java.util.Map;
+
+@RestController
+@RequestMapping("/files")
+public class FileController {
+
+    private FileService fileService;
+    private ObjectMapper objectMapper;
+    private FileController exportRecipeService;
+    private final static String STORE_FILE_NAME = "recipes";
+    private Map<Integer, Recipe> recipes = new HashMap<>();
+
+    public FileController(FileService fileService) {
+        this.fileService = fileService;
+    }
+
+    @GetMapping("/exportRec")
+    public ResponseEntity<InputStreamResource> downloadDataFile() throws FileNotFoundException {
+        File file = fileService.getDataFile();
+        if (file.exists()) {
+            InputStreamResource resource = new InputStreamResource(new FileInputStream(file));
+            return ResponseEntity.ok()
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .contentLength(file.length())
+                    .header(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=\"recipeLog.json\"")
+                    .body(resource);
+
+        } else {
+            return ResponseEntity.noContent().build();
+        }
+    }
+
+    @PostMapping(value = "/importRec", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+    public ResponseEntity<Void> uploadDataFile(@RequestParam MultipartFile file) {
+        fileService.cleanDataFile();
+        File dataFile = fileService.getDataFile();
+
+        try (FileOutputStream fos = new FileOutputStream(dataFile)) {
+            IOUtils.copy(file.getInputStream(), fos);
+            return ResponseEntity.ok().build();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+    }
+
+    @PostMapping(value = "/importIng", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+    public ResponseEntity<Void> uploadDataFile1(@RequestParam MultipartFile file) {
+        fileService.cleanDataFile1();
+        File dataFile = fileService.getDataFile1();
+
+        try (FileOutputStream fos1 = new FileOutputStream(dataFile)) {
+            IOUtils.copy(file.getInputStream(), fos1);
+            return ResponseEntity.ok().build();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+    }
+
+    //Получение всех рецептов из приложения
+    @Operation(
+            summary = "Загружаем список рецептов в формате txt"
+    )
+    @GetMapping("/exportAllRecipes")
+    public void exportAllRecipes(PrintWriter writer) {
+        for (Recipe recipe : this.recipes.values()) {
```

Comment:
Все рецепты мы получим из сервиса:
```suggestion
        for (Recipe recipe : recipeService.getAllRecipe().values()) {
```